lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -1575,22 +1579,27 @@ migrateWallet ctx (ApiT wid) migrateData = do
 -- the number of addresses in the specified address list, addresses will be
 -- recycled in order of their appearance in the original list.
 assignMigrationAddresses
-    :: [Address]
+    :: forall input output change .
```

Comment:
I wonder why to introduce type parametrization and use type equality (very concrete like `output ~TxOut`) rather than stick to what we had?

Reply:
this is one place, but this technique was adopted in many places in this PR

Reply:
> I wonder why to introduce type parametrization and use type equality (very concrete like output ~TxOut) rather than stick to what we had?

This is an attempt to improve readability.

Otherwise, we'd have to write things like:
```hs
UnsignedTx (TxIn, TxOut, NonEmpty DerivationIndex) TxOut (TxChange (NonEmpty DerivationIndex))
```
Which gets repetitive very quickly.

See `mkApiCoinSelection` for an example where the `input`, `output` and `change` types are also reused by inner function signatures.